server/channels/jobs/batch_report_worker.go

```diff
@@ -0,0 +1,149 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package jobs
+
+import (
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/mattermost/mattermost/server/public/model"
+	"github.com/mattermost/mattermost/server/public/shared/mlog"
+	"github.com/mattermost/mattermost/server/public/shared/request"
+	"github.com/mattermost/mattermost/server/v8/channels/store"
+	"github.com/pkg/errors"
+)
+
+type BatchReportWorkerAppIFace interface {
+	SaveReportChunk(format string, prefix string, count int, reportData []model.ReportableObject) *model.AppError
+	CompileReportChunks(format string, prefix string, numberOfChunks int, headers []string) *model.AppError
+	SendReportToUser(rctx request.CTX, userID string, jobId string, format string) *model.AppError
+	CleanupReportChunks(format string, prefix string, numberOfChunks int) *model.AppError
+}
+
+type BatchReportWorker[T BatchReportWorkerAppIFace] struct {
```

Comment:
Why is a generic being used here? Can't `app` just be of type `BatchReportWorkerAppIFace`?

Reply:
I was experimenting with using generics before, must have left this one. Will remove.

Reply:
We allow jobs that use the `batch_report_worker` to add their own methods to the interface, since it appears we define these interfaces separately.

To be honest, I was always kind of confused why we did that to begin with, and not just use `AppIFace` everywhere.

What is the actual correct way of doing this such that other jobs can still use the passed in `app` parameter here?

Reply:
I'd be fine with `AppIFace` over generics in this case, but if we wanted to avoid that, I suppose the more specific worker would need to define its own `app` and interface which is just a superset of the one this worker requires (and for which it maintains its own `app` as well).

Reply:
Let me try and make that work, as I think that is cleaner. As an aside, I really like generics :P